<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CodeGenerationDialogUi</class>
 <widget class="QDialog" name="CodeGenerationDialogUi">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>600</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>400</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Code Generation</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="KMessageWidget" name="topMessageWidget"/>
   </item>
   <item>
    <widget class="QLabel" name="titleLabel">
     <property name="text">
      <string>Select physical entities for code generation:</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="searchGroup">
     <property name="title">
      <string/>
     </property>
     <property name="flat">
      <bool>true</bool>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <property name="spacing">
       <number>6</number>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QLabel" name="searchTxt">
        <property name="text">
         <string>Search:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="searchValue"/>
      </item>
      <item>
       <widget class="QPushButton" name="searchGoToNextBtn">
        <property name="text">
         <string>Next</string>
        </property>
        <property name="icon">
         <iconset resource="../resources.qrc">
          <normaloff>:/icons/down</normaloff>:/icons/down</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="searchGoToPrevBtn">
        <property name="text">
         <string>Previous</string>
        </property>
        <property name="icon">
         <iconset resource="../resources.qrc">
          <normaloff>:/icons/up</normaloff>:/icons/up</iconset>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="selectAllCheckbox">
     <property name="text">
      <string>Select all</string>
     </property>
     <property name="tristate">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QTreeView" name="physicalEntitiesTree"/>
   </item>
   <item>
    <layout class="QFormLayout" name="formLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Code generation script:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QComboBox" name="availableScriptsCombobox"/>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Output directory:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLineEdit" name="outputDirInput"/>
       </item>
       <item>
        <widget class="QPushButton" name="findOutputDirBtn">
         <property name="text">
          <string>Find...</string>
         </property>
         <property name="icon">
          <iconset resource="../resources.qrc">
           <normaloff>:/icons/find</normaloff>:/icons/find</iconset>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="runCancelGroup">
     <property name="title">
      <string/>
     </property>
     <property name="flat">
      <bool>true</bool>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_5">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="cancelBtn">
        <property name="minimumSize">
         <size>
          <width>100</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>Cancel</string>
        </property>
        <property name="icon">
         <iconset resource="../resources.qrc">
          <normaloff>:/icons/cancel</normaloff>:/icons/cancel</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="runCodeGenerationBtn">
        <property name="minimumSize">
         <size>
          <width>150</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>Run</string>
        </property>
        <property name="icon">
         <iconset resource="../resources.qrc">
          <normaloff>:/icons/build</normaloff>:/icons/build</iconset>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="statusGroup">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QProgressBar" name="progressBar">
        <property name="value">
         <number>0</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="openOutputDir">
        <property name="text">
         <string>Open output directory</string>
        </property>
        <property name="icon">
         <iconset resource="../resources.qrc">
          <normaloff>:/icons/open-generated-code</normaloff>:/icons/open-generated-code</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="logsLabel">
        <property name="text">
         <string>Code generation logs:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="codeGenLogsTextArea">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>120</height>
         </size>
        </property>
        <property name="undoRedoEnabled">
         <bool>false</bool>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KMessageWidget</class>
   <extends>QFrame</extends>
   <header location="global">kmessagewidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resources.qrc"/>
 </resources>
 <connections/>
</ui>
